src/main/java/com/cachinginapp/enterprise/CachingInController.java

```diff
@@ -55,9 +51,12 @@ public List<Cache> fetchAllCaches(){
         return cacheService.fetchAll();
     }
 
-    @GetMapping("/cache/id/")
-    public ResponseEntity fetchCacheById(@PathVariable("id") String id){
-        return new ResponseEntity(HttpStatus.OK);
+    @GetMapping("/cache/{id}/")
+    public ResponseEntity fetchCacheById(@PathVariable("id") int id){
+        Cache foundCache = cacheService.fetchCacheById(id);
+        HttpHeaders headers = new HttpHeaders();
+        headers.setContentType(MediaType.APPLICATION_JSON);
+        return new ResponseEntity(foundCache, headers, HttpStatus.OK);
```

Comment:
changed id to int to match DTO & added headers & altered endpoint to /cache/{id}/